model.py

```diff
@@ -6219,33 +6219,36 @@ def assign_to_genre(self):
             genre, was_new = Genre.lookup(_db, genredata.name, True)
         else:
             genre = None
+
+        # Create a shorthand way of referring to this Subject in log
+        # messages.
+        parts = [self.type, self.identifier, self.name]
+        shorthand = ":".join(x for x in parts if x)
+
         if genre != self.genre:
             log.info(
-                "%s:%s genre %r=>%r", self.type, self.identifier,
-                self.genre, genre
+                "%s genre %r=>%r", shorthand, self.genre, genre
```

Comment:
I made this change to show identifier and/or name, rather than just identifier, because BISAC clasisfications have a name but might not have an identifier.